app/tests/infrastructure/test_exchange_repository.py

```diff
@@ -168,3 +174,43 @@ def test_get_by_download_token_should_return_none_if_no_exchange_exists(self, mo
 
         # Then
         assert actual_exchange is None
+
+
+@pytest.mark.usefixtures("setup_dynamodb_container")
+@pytest.mark.usefixtures("get_dynamodb_table")
+class TestExchangeRepositoryActionsBasedOnTime:
+    @patch('hallebarde.infrastructure.exchange_repository._get_dynamodb_table')
+    def test_get_before_time_should_return_only_exchanges_created_before_this_time(self, mock_get_table,
+                                                                                   get_dynamodb_table, an_exchange,
+                                                                                   generate_old_exchange):
+        # Given
+        mock_get_table.return_value = get_dynamodb_table
+        time_before = datetime.now(timezone.utc) - timedelta(days=2)
+
+        old_exchange = generate_old_exchange(days_before=3)
+        exchange_repository.save(an_exchange)
+        exchange_repository.save(old_exchange)
+
+        # When
+        actual_exchanges = exchange_repository.get_before_time(time_before)
+
+        # Then
+        assert actual_exchanges == [old_exchange]
+
+    @patch('hallebarde.infrastructure.exchange_repository._get_dynamodb_table')
+    def test_get_before_time_should_handle_cases_with_more_exchanges_to_retrieve_than_limit_in_dynamodb(self,
+                                                                                                        mock_get_table,
+                                                                                                        get_dynamodb_table,
+                                                                                                        generate_old_exchange):
+        # Given
+        mock_get_table.return_value = get_dynamodb_table
+        week_before = datetime.now(timezone.utc) - timedelta(days=7)
+
+        for i in range(120):
+            exchange_repository.save(generate_old_exchange(days_before=8))
```

Comment:
I generally use `_` for variable that I won't use over a loop.

Also, a fancy way to write this kind of loops I know is:
```suggestion
        map(lambda x: exchange_repository.save(generate_old_exchange(days_before=8)), range(120))          
```

Bref, :D

Reply:
Yeah I do agree, though I find this way of writing it less… explicit? or obvious? Do you think we should use it @Mehdi-H ?

I changed the variable name

Reply:
Yeah, I don't this that the map(lambda...) is easily readable. But I think that the `_` is pretty pythonic